### ISSUE TITLE
Incorrect dot product fallback

### ISSUE THREAD

**Issue by juliohm** [JuliaMath/Polynomials.jl#397]
The fallback is producing results that are not mathematically correct:

```julia
julia> using Polynomials
julia> using LinearAlgebra
julia> p = Polynomial([0,1])
julia> p ⋅ p
1
```

**Comment by jverzani** [comment]
At one time, we had `p ⋅ p` mean multiplication, now it falls back to `LinearAlgebra.dot` for arbitrary iterables. It should be that for vector `v` `P(v) ⋅ P(v) == v ⋅ v`.

**Comment by juliohm** [comment]
So you are not implementing the dot product as the integral `\int_{-\infty}^{\infty} f(t) * conj(g(t)) dt`?

**Comment by jverzani** [comment]
No, right now we don't have it implemented for the `AbstractPolynomial` type, so it fallsback to the generic for iterables. In `SpecialPolynomials`, for orthogonal polynomials, we have something defined like that internally which should be exposed through `dot`. I'll definitely add that in.

**Comment by juliohm** [comment]
This should probably get the `bug` label.

**Comment by jverzani** [comment]
Is it a bug? Or does it need a special method. Right now the fall back is to the `dot` method for two iterables which will error for polynomials of different degree. For orthogonal polynomials, we could use the integral over `[-1,1]`, but in general the integral over `(-oo, oo)` will always yield an infinite answer, so that isn't the best. 

So would having
```
Polynomials.dot(p::AbstractPolynomial, q::AbstractPolynomial) = throw(ArgumentError("..."))
```
and
```
SpecialPolynomials.dot(p::AbstractOrthogonalPolynomial, q::AbstractOrthogonalPolynomial) = integrate(p * conj(q), domain(p)...)
```
suffice?

**Comment by jverzani** [comment]
closed by #563
